### PR TITLE
Update merge master acc

### PR DESCRIPTION
**Pull request by njgheorghita**
### What was wrong?
The `MasterAccumulator` struct resulted in an invalid ssz encoding/decoding.

### How was it fixed?
Removed the redundant nested types from `MasterAccumulator`

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [x] Add entry to the [release notes](https://github.com/ethereum/trin/blob/master/newsfragments/README.md) (may forgo for trivial changes)
- [x] Clean up commit history

### CODE REVIEW

File: trin-core/src/types/validation.rs [hunk 1]
```diff
@@ -13,9 +13,9 @@ use crate::{
     utils::provider::TrustedProvider,
 };
 
-pub const MERGE_BLOCK_NUMBER: u64 = 15_537_394u64;
+pub const MERGE_BLOCK_NUMBER: u64 = 15_537_393u64;
```
Comment (carver) [inline trin-core/src/types/validation.rs:16]: This change (and the other parts where the accumulator ends one header earlier) seems like it belongs in its own commit separate from the fix to the SSZ structure.

File: trin-core/src/types/accumulator.rs [hunk 2]
```diff
@@ -23,23 +23,23 @@ use crate::utils::bytes::{hex_decode, hex_encode};
 pub const EPOCH_SIZE: usize = 8192;
 
 /// Max number of epochs - 2 ** 17
-const MAX_HISTORICAL_EPOCHS: usize = 131072;
+const _MAX_HISTORICAL_EPOCHS: usize = 131072;
```
Comment (ogenev) [inline trin-core/src/types/accumulator.rs:26]: What is the reason for prefixing this with `_` instead of deleting the constant?

**Review by ogenev — APPROVED** [review]
:boat:

File: trin-core/src/types/accumulator.rs (hunk 2 above)
Comment (njgheorghita) [reply trin-core/src/types/accumulator.rs]: We reference the constant in the line below, so I thought to leave it in to have the actual int as a reference. But yeah, it is unnecessarily compiled, so I've commented it out
